src/Microsoft.Azure.ServiceBus/Core/MessageSender.cs

```diff
@@ -90,6 +92,40 @@ public class MessageSender : ClientEntity, IMessageSender
             }
         }
 
+        private async Task<Message> ProcessMessage(Message message)
+        {
+            if (this.RegisteredPlugins == null)
+            {
+                return message;
+            }
+
+            var processedMessage = message;
+            foreach (var plugin in this.RegisteredPlugins)
+            {
+                try
+                {
+                    processedMessage = await plugin.BeforeMessageSend(message).ConfigureAwait(false);
+                }
+                catch (Exception ex) when (ex is ServiceBusPluginException pluginException && pluginException.ShouldCompleteOperation)
+                {
+                    continue;
+                }
+            }
+            return processedMessage;
+        }
+
+        private async Task<IList<Message>> ProcessMessages(IList<Message> messageList)
```

Comment:
>private async Task<IList<Message>> ProcessMessages(IList<Message> messageList) [](start = 8, length = 78)

return quick if no plugins found instead of iterating through the messages